File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService",function ($http) {

    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }

    this.findPage=function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }

    this.findOne=function (id) {
        return $http.post('../brand/findOne.do?id=' + id, $scope.entity);
    }

    this.add=function (entity) {
        return  $http.post('../brand/add.do', entity)
    }

    this.update=function (entity) {
        return  $http.post('../brand/update.do', entity)
    }

    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids)
    };

    this.search=function (page,size,searchEntity) {
        return  $http.post('../brand/search.do?page=' + page + '&size=' + size,searchEntity)
    }


    /*
           下拉列表数据  因为展示的是品牌的下拉列表,所以在brandService.js中发送请求,
         在typeTemplateController.js中调取服务.

     */
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;


import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {


    /**
     * 查询品牌的所有信息
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum 当前的页数
     * @param pageSize 每页的记录数
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);


    /**
     * 添加品牌信息
     * @param brand
     */
    public void add(TbBrand brand);



    /**
     * 根据id查询品牌信息
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);



    /**
     * 修改品牌信息
     * @param brand
     */
    public void update(TbBrand brand);




    /**
     * 根据id删除品牌信息
     * @param ids
     */
    public void delete(Long[] ids);


    /**
     * 模糊查询,并进行分页
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);


    /**
     * 返回下拉列表数据
     * @return
     */
    public List<Map> selectOptionList();







}
